### PR TITLE
Use mainline driver for WiFi on Flat S 2022 (REVPI-2882)

### PR DESCRIPTION
**Pull request by l1k**

### CODE REVIEW

File: include/linux/mmc/sdio_ids.h [hunk 1]
```diff
@@ -98,6 +98,7 @@
 #define SDIO_DEVICE_ID_MARVELL_8997_BT		0x9142
 #define SDIO_DEVICE_ID_MARVELL_8977_WLAN	0x9145
 #define SDIO_DEVICE_ID_MARVELL_8977_BT		0x9146
+#define SDIO_DEVICE_ID_MARVELL_8978_WLAN	0x9159
```
Comment (iluminat23) [inline include/linux/mmc/sdio_ids.h]: I would place the definition after `SDIO_DEVICE_ID_MARVELL_8987_BT` as the defines seem to be sorted by the value not the define/chip name.

**Comment by iluminat23** [comment]
Please add your Signed-off-by to the commits back ported from upstream. This is the proper behavior for handling patches.
* 7c067f58e6d95f1c84a1b258b114fb3838f5df4f
* 680b9d4cd798eaeb90580e9227f15c42db31751f


The patch https://github.com/RevolutionPi/linux/pull/141/commits/8dd8682c40ca579993336acc44b823b26b0f039f has a `Signed-off-by: Sherry Sun <sherry.sun@nxp.com>` which is misleading. For me it seems there is no relevant code which was authored by Sherry Sun to justify adding the DCO here. As the DCO would imply Sherry Sun did actually worked on the patch. From my point of view you did enough when you added the link to the commit in the nxp repo.

**Comment by iluminat23** [comment]
If you want to upstream this, you might want to split up this commit: 8abe48a310ed03052153dd94f69d86cd7191c860
Sometimes they want to have (devicetree-)documentation changes in an extra commit.

**Comment by l1k** [comment]
All of these patches are now upstream, so should be ready to go into the revpi-5.10 branch.

**Review by iluminat23 — APPROVED** [review]

**Review by nbuchwitz — APPROVED** [review]
Package builds for arm and arm64 were successful. Setup of the packages on a Flat / Flat S worked -> LGTM

**Pull request merged**